### PR TITLE
[MAISTRA-750] Unify templates and charts dir

### PR DESCRIPTION
**Pull request by jwendell**
And make /usr/local/share/istio-operator the default location

### CODE REVIEW

File: tmp/build/download-charts.sh [hunk 1]
```diff
@@ -37,13 +37,11 @@ function retrieveIstioRelease() {
...
-      echo "downloading Istio Release: ${ISTIO_URL}"
-      cd "${RELEASES_DIR}"
-      curl -LO "${ISTIO_URL}"
-    )
-  fi
+  (
+    echo "downloading Istio Release: ${ISTIO_URL}"
+    cd "${RELEASES_DIR}"
+    curl -LO "${ISTIO_URL}"
+  )
```
Comment (knrc) [inline tmp/build/download-charts.sh]: It looks like this change snuck into the PR, it's not related to the task at hand

**Review by knrc — CHANGES_REQUESTED** [review]

File: tmp/build/Dockerfile [hunk 2]
```diff
@@ -3,6 +3,6 @@ FROM registry.access.redhat.com/ubi8/ubi-minimal:latest
 ARG build_type=maistra
 
 ADD tmp/_output/bin/istio-operator /usr/local/bin/istio-operator
-ADD tmp/_output/templates/ /usr/share/istio-operator/default-templates
+ADD tmp/_output/templates/ /usr/local/share/istio-operator/default-templates
 ADD manifests-${build_type} /manifests
-COPY tmp/_output/helm/istio-releases/istio-1.1.0 /etc/istio-operator/1.1.0/helm
+COPY tmp/_output/helm/istio-releases/istio-1.1.0 /usr/local/share/istio-operator/helm
```
Comment (luksa) [inline tmp/build/Dockerfile]: I think we need to keep the version in the directory name, as we'll eventually have multiple versions in there.

**Review by luksa — COMMENTED** [review]
Can we also reorganize the paths in `tmp/_output` so that it contains a `resources` directory with all required subdirectories (`templates`, `default-templates` and `helm`). This will allow us to run the operator locally with `--resourceDir tmp/_output/resources/`.

File: tmp/build/Dockerfile (hunk 2 above)
Comment (jwendell) [reply tmp/build/Dockerfile]: @luksa  Do you think we should do this now or can we leave that for the future, when we proper implement it?

File: tmp/build/download-charts.sh (hunk 1 above)
Comment (jwendell) [reply tmp/build/download-charts.sh]: heh, I had this change locally and got caught by git commit -a :/
Fixed, thanks.

**Comment by jwendell** [comment]
> Can we also reorganize the paths in `tmp/_output` so that it contains a `resources` directory with all required subdirectories (`templates`, `default-templates` and `helm`). This will allow us to run the operator locally with `--resourceDir tmp/_output/resources/`.

Done. I wanted to make this as a not intrusive change (not touching download-charts), so, I'm just copying the files to the right location after patch-charts is run.

File: tmp/build/Dockerfile (hunk 2 above)
Comment (luksa) [reply tmp/build/Dockerfile]: Since we had the version in so far, I'd say it's best to keep it.

**Comment by jwendell** [comment]
@luksa ok, I went with the version being a subdir of helm dir:

```
resources
resources/default-templates
resources/helm
resources/helm/1.1.0
```

Right now the version 1.1.0 is hard coded (that has not changed from what we have now) everywhere until MAISTRA-766 is implemented.

**Comment by jwendell** [comment]
I've also updated https://github.com/Maistra/rpms/pull/65 to reflect this change

**Review by knrc — APPROVED** [review]

**Comment by luksa** [comment]
@jwendell Don't we need to version the templates too? So they should be under 1.1.0, right?

**Review by luksa — APPROVED** [review]

**Pull request merged**